sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -297,7 +297,7 @@ def initialize(context, parent_context, parent_span, name, kind, parent_span_id,
           @total_recorded_events = 0
           @total_recorded_links = links&.size || 0
           @total_recorded_attributes = attributes&.size || 0
-          @attributes = attributes.nil? ? nil : Hash[attributes] # We need a mutable copy of attributes.
+          @attributes = attributes
```

Comment:
Sometimes, this line was calling an already-`dup`ed Hash, which had been created by `attributes.merge(result.attributes)`. Instead of duping it here, I added a `.dup` call to the calling code in `tracer_provider.rb`. That way, we'd only create new Hashes when we _really_ needed them.  

I didn't find any other callers to this class -- was that a safe substitution?